src/lib/routes/admin-api/project/api-token.ts

```diff
@@ -0,0 +1,214 @@
+        {
+            apiTokenService,
+            accessService,
+            proxyService,
+            openApiService,
+        }: Pick<
+            IUnleashServices,
+            | 'apiTokenService'
+            | 'accessService'
+            | 'proxyService'
+            | 'openApiService'
+        >,
+    ) {
+        super(config);
+        this.apiTokenService = apiTokenService;
+        this.accessService = accessService;
+        this.proxyService = proxyService;
+        this.openApiService = openApiService;
+        this.logger = config.getLogger('project-api-token-controller.js');
+
+        this.route({
+            method: 'get',
+            path: PATH,
+            handler: this.getProjectApiTokens,
+            permission: READ_PROJECT_API_TOKEN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Projects'],
+                    operationId: 'getProjectApiTokens',
+                    responses: {
+                        200: createResponseSchema('apiTokensSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'post',
+            path: PATH,
+            handler: this.createProjectApiToken,
+            permission: CREATE_PROJECT_API_TOKEN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Projects'],
+                    operationId: 'createProjectApiToken',
+                    requestBody: createRequestSchema('createApiTokenSchema'),
+                    responses: {
+                        201: resourceCreatedResponseSchema('apiTokenSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'delete',
+            path: PATH_TOKEN,
+            handler: this.deleteProjectApiToken,
+            acceptAnyContentType: true,
+            permission: DELETE_PROJECT_API_TOKEN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Projects'],
+                    operationId: 'deleteProjectApiToken',
+                    responses: {
+                        200: emptyResponse,
+                    },
+                }),
+            ],
+        });
+    }
+
+    async getProjectApiTokens(
+        req: IAuthRequest,
+        res: Response<ApiTokensSchema>,
+    ): Promise<void> {
+        const { user } = req;
+        const { projectId } = req.params;
+        const projectTokens = await this.accessibleTokens(user, projectId);
+        this.openApiService.respondWithValidation(
+            200,
+            res,
+            apiTokensSchema.$id,
+            { tokens: serializeDates(projectTokens) },
+        );
+    }
+
+    async createProjectApiToken(
+        req: IAuthRequest,
+        res: Response<ApiTokenSchema>,
+    ): Promise<any> {
+        const createToken = await createApiToken.validateAsync(req.body);
```

Comment:
This validation is using joi, which is the old way of validating data. Since we swapped to openAPI validation, we should use that instead so we don't accrue more technical debt.

Reply:
makes sense